libp2p/Session.cpp

```diff
@@ -100,16 +82,18 @@ int Session::rating() const
     return m_peer->m_rating;
 }
 
-bool Session::readPacket(uint16_t _capId, PacketType _packetType, RLP const& _r)
+bool Session::readPacket(uint16_t _capId, unsigned _packetType, RLP const& _r)
 {
     m_lastReceived = chrono::steady_clock::now();
-    clog(VerbosityTrace, "net") << "-> " << _packetType << " " << _r;
+    clog(VerbosityTrace, "net") << "Received " << capabilityPacketTypeToString(_packetType) << "("
+                                << _packetType << ") from " << m_info.id << "@"
```

Comment:
I'm not sure about logging everywhere in Session class `<< m_info.id << "@" << m_socket->remoteEndpoint();`

Here `LOG_SCOPED_CONTEXT` is supposed to add a prefix of id+clientVersion for each log message inside log handlers. Isn't at least `m_info.id` duplicated in logs this way?

I suspect that sometimes IDs could be different, like when `BlockChainSync` code is called inside the handler of the message from one peer, and this code sends a message to another peer.
In this case, maybe it's better to remove `LOG_SCOPED_CONTEXT` stuff now? Though then clientVersion of the peers won't as visible in logs as now, it seems it's useful, too... What do you think?

Reply:
I like having the client information in the log messages, though I don't have a concrete example of when it's helped me debug something yet. I think that prefixing each log message with the node id + client information makes the log messages a little counter-intuitive, for example, the following message could be logged when we receive a Pong p2p capability message from a node but the message reads as if 865a6325 received the pong:

`p2p p2pcap 865a6325…|Parity-Ethereum/v2.2.9-stable-5d5b372-20190203/x86_64-linux-gnu/rustc1.31.1 Receiving Pong`

An option is to remove the LOG_SCOPED_CONTEXT and to instead log the client information at the end of each message e.g. 

`TRACE 05-07 21:03:32 p2p  p2pcap Pong to ##865a6325…@52.176.100.77:30303 (Parity-Ethereum/v2.2.9-stable-5d5b372-20190203/x86_64-linux-gnu/rustc1.31.1)`

I think this is more readable, though it makes the log statements themselves more cumbersome (as you called out in my RLPXHandshake logging changes). Perhaps we could also create a logging wrapper function such as:

```
inline void logMessage(char const* _message, char const* _channel, Verbosity _verbosity)
{
      clog(_verbosity, _channel) << _message << " (" << m_info.clientVersion << "|" << m_info.id << "|" << m_socket->remoteEndpoint() << ")";
}
```

Thoughts?

Reply:
I think I dislike repeating clientVersion many times, it bloats the log. If we output it once, say for Hello message, it should be easy enough to find it in the log for particular NodeID, when we need it.

Reply:
Another thing: if we repeat this `<< m_info.id << "@" << m_socket->remoteEndpoint();` in every message and these values are constant, can we create a logger for this, that adds them as a Context attribute, similar to what you did in `RLPXHandshake` ?

This and removing `LOG_SCOPED_CONTEXT` can be done as a separate PR.

Reply:
The downside to removing `LOG_SCOPED_CONTEXT` would be that we won't see this context for the messages coming from `BlockChainSync`, not sure how useful it is. Maybe it's fine to remove, as all network handling happens sequentially in one thread and we should be able to find this context earlier in the log.

Reply:
> 
> 
> I think I dislike repeating clientVersion many times, it bloats the log. If we output it once, say for Hello message, it should be easy enough to find it in the log for particular NodeID, when we need it.

Yes, but the Hello might be long in the past so it might not be accessible in the console log output anymore. This might be okay though since we also periodically log connected peers, though only at trace level: https://github.com/ethereum/aleth/blob/7ecf42b4e4afd9fb5c323b2338c5ac60d6f41dea/libp2p/Host.cpp#L868

`Host::m_detailLogger` is defined here: https://github.com/ethereum/aleth/blob/7ecf42b4e4afd9fb5c323b2338c5ac60d6f41dea/libp2p/Host.h#L382

Maybe we should change the verbosity level of this log message to debug level so that it's more readily available for debugging purposes?

Reply:
> e repeat this `<< m_info.id << "@" << m_socket->remoteEndpoint();` in every message and these values are constant, can we create a logger for this, that adds them as a Context attribute...

@gumb0 Is there a way to add the context to the end of a log message? I think including remote endpoint information at the beginning of a log message makes the log a little counter-intuitive.

Reply:
> Is there a way to add the context to the end of a log message?

Not right now, but I think it's simple enough to add this. We could rename `Context` attribute to `Prefix`, then add another `Suffix` one, and your logger would fill the `Suffix`

The changes will be need only here https://github.com/ethereum/aleth/blob/4e63cf289c2d1902ece63ce53545b25bd0257bae/libdevcore/Log.cpp#L51
and when we output it in formatter here
https://github.com/ethereum/aleth/blob/4e63cf289c2d1902ece63ce53545b25bd0257bae/libdevcore/Log.cpp#L99-L100